cmake/module/GetTargetInterface.cmake

```diff
@@ -32,7 +32,18 @@ endfunction()
 function(get_target_interface var config target property)
   get_target_property(result ${target} INTERFACE_${property})
   if(result)
-    evaluate_generator_expressions(result "${config}")
+    # On systems where pthread functionality is not provided by
+    # the C library implementation, the CMake FindThreads module
+    # sets the Threads::Threads target's compile options to
+    # generator expressions that evaluate to `-pthread` in this
+    # project.
+    # To improve the readability of the configuration summary,
+    # we skip these generator expressions.
+    if(${target} STREQUAL "Threads::Threads" AND ${property} STREQUAL "COMPILE_OPTIONS")
```

Comment:
not blocking: if an upstream change in CMake ever sets different `COMPILE_OPTIONS` for `Threads::Threads` (probably unlikely) we would print something wrong in the flags summary,

What do you think of this as an alternative?

```diff
--- a/cmake/module/GetTargetInterface.cmake
+++ b/cmake/module/GetTargetInterface.cmake
@@ -20,6 +20,10 @@ function(evaluate_generator_expressions list config)
       if(NOT CMAKE_MATCH_1 STREQUAL config)
         list(APPEND result ${CMAKE_MATCH_2})
       endif()
+    elseif(token MATCHES "\\$<\\$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:SHELL:[^>]+>")
+      # do nothing
+    elseif(token MATCHES "\\$<\\$<AND:\\$<NOT:\\$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>>,\\$<NOT:\\$<COMPILE_LANGUAGE:Swift>>>:([^>]+)>")
+      list(APPEND result ${CMAKE_MATCH_1})
```

Reply:
> if an upstream change in CMake ever sets different `COMPILE_OPTIONS` for `Threads::Threads` (probably unlikely)...

I would argue the opposite. It’s possible that more cases could be added where the `-pthread` flag should be filtered out using additional generator expressions. The last [modification](https://gitlab.kitware.com/cmake/cmake/-/commit/d7963aa9ee38bf26ab31433f2e7bfaff7ddf6c57) was made two years ago.

Reply:
I agree, but in that situation we would be back to printing generator expressions in the flags summary, which seems to me better than silently printing something that is not true.

Reply:
> I agree, but in that situation we would be back to printing generator expressions in the flags summary...

Unfortunately, it happens not only with unknown future changes but also with older yet supported CMake versions, such as 3.25, where `$<$<NOT:$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>>:-pthread>` is printed.